File: tvr-decoder/src/main/java/com/willcurrie/decoders/apdu/APDUCommand.java
package com.willcurrie.decoders.apdu;

public enum APDUCommand {
    Select("00A4"),
    ReadRecord("00B2"),
    GetProcessingOptions("80A8"),
    GenerateAC("80AE"),
    ComputeCryptographicChecksum("802A"),
    GetData("80CA"),
    ExternalAuthenticate("0082");

    private String firstTwoBytes;

    private APDUCommand(String firstTwoBytes) {
        this.firstTwoBytes = firstTwoBytes;
    }

    public String getFirstTwoBytes() {
        return firstTwoBytes;
    }

    public static APDUCommand fromHex(String hex) {
        APDUCommand[] values = values();
        for (APDUCommand value : values) {
            if (value.getFirstTwoBytes().equals(hex)) {
                return value;
            }
        }
        return null;
    }
}

File: src/main/java/io/github/binaryfoo/decoders/apdu/APDUCommand.kt
package io.github.binaryfoo.decoders.apdu

import kotlin.collections.firstOrNull

enum class APDUCommand constructor(val firstTwoBytes: String) {
    Select("00A4"),
    ReadRecord("00B2"),
    ReadBinary("00B0"),
    GetProcessingOptions("80A8"),
    GenerateAC("80AE"),
    ComputeCryptographicChecksum("802A"),
    GetData("80CA"),
    ExternalAuthenticate("0082"),
    InternalAuthenticate("0088"),
    Verify("0020"),
    GetChallenge("0084"),
    PutData("04DA");

    companion object {
        fun fromHex(hex: String): APDUCommand? = values().firstOrNull { it.firstTwoBytes == hex }
    }
}
